### PR TITLE
#36: Repository upload with reactive interface

### PR DESCRIPTION
**Pull request by Nimmaster**
issue #36 BTW reactive approach makes uploading lifecycle clearer.
`Deployer` class javadoc has a qulice suppression, explained in `AetherRepository`.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#45](https://github.com/artipie/maven-adapter/issues/45) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/e9f68eb7-a34c-4a29-ba21-2f1ed19c0348); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by fanifieiev** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/maven-adapter/issues/45#issuecomment-589243416))

@fanifieiev The user @fanifieiev[/z](https://www.0crat.com/u/fanifieiev) resigned from [#45](https://github.com/artipie/maven-adapter/issues/45), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/maven-adapter/issues/45#issuecomment-589243416))

@fanifieiev Job refused in 8 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by Nimmaster** [comment]
@g4s8 pls assign a reviewer

**Comment by paulodamaso** [comment]
@g4s8 please assign me

**Comment by g4s8** [comment]
@0crat assign @paulodamaso

**Comment by 0crat** [comment]
> @0crat assign @paulodamaso  ([here](https://github.com/artipie/maven-adapter/issues/45#issuecomment-589624538))

@g4s8 This pull request [#45](https://github.com/artipie/maven-adapter/issues/45) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/cb2be02e-c280-436c-ba25-2d78eaacc3d1); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/java/com/artipie/maven/aether/AstoTransporter.java [hunk 1]
```diff
@@ -69,17 +76,21 @@ public void implPeek(final PeekTask task) throws Exception {
 
     @Override
     public void implGet(final GetTask task) throws Exception {
+        LOG.debug("implGet start {}", task.getLocation().toString());
```
Comment (paulodamaso) [inline src/main/java/com/artipie/maven/aether/AstoTransporter.java]: @Nimmaster Do we need all these log / debugging lines in the code? We should find bugs and fix code by writing tests, not debugging and reading debug code lines

Comment (Nimmaster) [reply src/main/java/com/artipie/maven/aether/AstoTransporter.java]: We can do both

Comment (paulodamaso) [reply src/main/java/com/artipie/maven/aether/AstoTransporter.java]: @g4s8 I just have this concern, if it's ok to you, it's ok to merge then

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**